### PR TITLE
support on update in create table an update

### PR DESCRIPTION
**Pull request by iamlinjunhong**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #3603 

## What this PR does / why we need it:

### CODE REVIEW

**Review by aunjgr — APPROVED** [review]

File: pkg/sql/parsers/dialect/mysql/mysql_sql.y [hunk 1]
```diff
@@ -4725,10 +4727,19 @@ column_attribute_elem:
     {
         $$ = tree.NewAttributeCheck($4, $6, $1)
     }
-// |   ON UPDATE function_call_nonkeyword
-//     {
-//         $$ = tree.NewAttributeOnUpdate($3)
-//     }
+|   ON UPDATE name_datetime_precision datetime_precision_opt
```
Comment (fengttt) [inline pkg/sql/parsers/dialect/mysql/mysql_sql.y:4730]: Why we limit update on date/time?    In theory, any function, right?

File: pkg/sql/plan/build_util.go [hunk 2]
```diff
@@ -165,6 +166,42 @@ func buildDefaultExpr(col *tree.ColumnTableDef, typ *plan.Type) (*plan.Default,
...
+	}
+
+	onUpdateExpr, err := makePlan2CastExpr(planExpr, typ)
+	if err != nil {
+		return nil, err
+	}
+
+	// try to calculate on update value, return err if fails
+	bat := batch.NewWithSize(0)
+	bat.Zs = []int64{1}
```
Comment (fengttt) [inline pkg/sql/plan/build_util.go:196]: This Zs is so strange.   Is this the right way to do things?   Should we init in within NewWithSize?

File: pkg/sql/parsers/dialect/mysql/mysql_sql.y (hunk 1 above)
Comment (fengttt) [reply pkg/sql/parsers/dialect/mysql/mysql_sql.y:4730]: OK, we leave it as it is for now.    Extend this feature later.

**Review by fengttt — APPROVED** [review]

File: pkg/sql/plan/build_util.go (hunk 2 above)
Comment (nnsgmsone) [reply pkg/sql/plan/build_util.go:196]: Here it simply means that a row of data exists

**Review by nnsgmsone — APPROVED** [review]

**Review by aressu1985 — APPROVED** [review]

**Pull request merged**